lib/index.js

```diff
@@ -98,11 +99,19 @@ shp.parseZip = async function (buffer, whiteList) {
       parsed = zip[name];
       parsed.fileName = name;
     } else {
+      const extensions = [];
+      for (key in zip) {
+        const extension = key.replace(name, '');
+        if (extension !== key) {
```

Comment:
i think you also need to do a check to make sure that extension starts with a '.'